### PR TITLE
Onboard Camel K Runtime fixups

### PR DESCRIPTION
**Pull request by ppalaga**
Some enhancements on top of https://github.com/apache/camel-quarkus/pull/5117

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by jamesnetherton** [comment]
Are we proposing to merge this for 3.2.0?

**Comment by ppalaga** [comment]
> Are we proposing to merge this for 3.2.0?

That's a good question. I think it would be very practical to have this in the LTS. Any opinion, @claudio4j?

**Comment by jamesnetherton** [comment]
> I think it would be very practical to have this in the LTS

I was thinking the reverse :slightly_smiling_face:. We are effectively done with 3.2.0, not sure I like introducing a change like this at this stage. Safer to defer for 3.3.0 IMO.

**Comment by ppalaga** [comment]
> I was thinking the reverse 🙂. We are effectively done with 3.2.0, not sure I like introducing a change like this at this stage. Safer to defer for 3.3.0 IMO.

I won't stand in a way then.

**Comment by zhfeng** [comment]
I'm also not confident to introduce such a big change in last minute before releasing `3.2.0`. I think @zbendhiba is preparing for the new release this week. WDYT?

**Comment by zbendhiba** [comment]
I was thinking of releasing 3.3.0 very soon after 3.2.0, so that we go back to the release cadence we used to have before. The 2 versions will be based on Camel 4, but 3.2.0 is an LTS. 
I also think it makes sense to have the experimental stuff for 3.3.0 and Camel k team will have quickly 3.3.0 to start experimenting stuff. is that makes sense for you @claudio4j  ?

**Comment by zhfeng** [comment]
@zbendhiba For releasing `3.3.0`, please make sure we have all changes in
- https://github.com/apache/camel-quarkus/pull/5190

**Comment by aldettinger** [comment]
With Camel-K 2.0, we should already have less coupling between Camel Quarkus and Camel-K ? So that should be ok to merge this pr in next 3.3.0 release ?

**Comment by ppalaga** [comment]
> It looks like camel-k-runtime tests are failing on windows build due to lack of Docker environment.

Thanks for having a look @zhfeng! I am away for a RH event today and I'll be able to address this only tomorrow.

**Comment by ppalaga** [comment]
866fe21: 

* Rebased
* Squashed all commits
* Added `skip-testcontainers-tests` profile to the new Camel K test module

```
        <profile>
            <id>skip-testcontainers-tests</id>
            <activation>
                <property>
                    <name>skip-testcontainers-tests</name>
                </property>
            </activation>
            <properties>
                <skipTests>true</skipTests>
            </properties>
        </profile>
```

**Comment by claudio4j** [comment]
> I was thinking of releasing 3.3.0 very soon after 3.2.0, so that we go back to the release cadence we used to have before. The 2 versions will be based on Camel 4, but 3.2.0 is an LTS. I also think it makes sense to have the experimental stuff for 3.3.0 and Camel k team will have quickly 3.3.0 to start experimenting stuff. is that makes sense for you @claudio4j ?

Sounds good to me, thanks for the feedback everyone.

File: integration-tests/camel-k-runtime/src/test/java/org/apache/camel/quarkus/k/it/RuntimeCustomizerTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import static io.restassured.RestAssured.given;
+import static org.apache.camel.util.CollectionHelper.mapOf;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Disabled
+//@TestProfile(RuntimeCustomizerTest.Profile.class)
+//@QuarkusTest
```
Comment (zbendhiba) [inline integration-tests/camel-k-runtime/src/test/java/org/apache/camel/quarkus/k/it/RuntimeCustomizerTest.java:33]: Same question here ?

File: integration-tests/camel-k-runtime/src/main/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+## ---------------------------------------------------------------------------
+
+#
+# Quarkus
+#
+#quarkus.log.console.enable = false
```
[the remark this answers is not in the document]
Comment (ppalaga) [reply integration-tests/camel-k-runtime/src/main/resources/application.properties]: Solved in 5c4c745

File: integration-tests/camel-k-runtime/src/test/java/org/apache/camel/quarkus/k/it/RuntimeCustomizerFallbackTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import static io.restassured.RestAssured.given;
+import static org.apache.camel.util.CollectionHelper.mapOf;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Disabled
+//@TestProfile(RuntimeCustomizerFallbackTest.Profile.class)
+//@QuarkusTest
```
[the remark this answers is not in the document]
Comment (ppalaga) [reply integration-tests/camel-k-runtime/src/test/java/org/apache/camel/quarkus/k/it/RuntimeCustomizerFallbackTest.java:33]: Thanks for noticing, @zbendhiba, I created a followup https://github.com/apache/camel-quarkus/issues/5235 for this and added `@Disabled("https://github.com/apache/camel-quarkus/issues/5235")` to the failing test.

File: integration-tests/camel-k-runtime/src/test/java/org/apache/camel/quarkus/k/it/RuntimeCustomizerTest.java (hunk 1 above)
Comment (ppalaga) [reply integration-tests/camel-k-runtime/src/test/java/org/apache/camel/quarkus/k/it/RuntimeCustomizerTest.java:33]: Thanks for noticing, @zbendhiba, I created a followup https://github.com/apache/camel-quarkus/issues/5235 for this and added `@Disabled("https://github.com/apache/camel-quarkus/issues/5235")` to the failing test.

**Review by zbendhiba — APPROVED** [review]

**Pull request merged**